eva-accession-clustering/src/test/java/uk/ac/ebi/eva/accession/clustering/batch/io/clustering_writer/RemappedVariantsClusteringWriterTest.java

```diff
@@ -339,137 +339,173 @@ public void test_variant_not_clustered_rs_id_exist() throws Exception {
-        assertEquals(Long.valueOf(3000000000L), expectedClusteredVariantOperationEntity.getSplitInto());
-        assertEquals(EventType.RS_SPLIT, expectedClusteredVariantOperationEntity.getEventType());
-        assertEquals("Due to RS split new accession id 3000000000 created for remapped accession 3000000006",
-                     expectedClusteredVariantOperationEntity.getReason());
+        List<SubmittedVariantEntity> submittedVariantEntityList = new ArrayList<>();
+        submittedVariantEntityList.add(submittedVariantEntityRemapped1);
+        submittedVariantEntityList.add(submittedVariantEntityRemapped3);
+        submittedVariantEntityList.add(submittedVariantEntityRemapped2);
+        submittedVariantEntityList.add(submittedVariantEntityRemapped4);
+        submittedVariantEntityList.add(submittedVariantEntityRemapped5);
+        submittedVariantEntityList.add(submittedVariantEntityRemapped6);
+        clusteringWriter.write(submittedVariantEntityList);
+
+        //assert clusteredVariantEntity
+        List<ClusteredVariantEntity> clusteredVariantEntities = mongoTemplate.findAll(ClusteredVariantEntity.class);
+        assertEquals(4, clusteredVariantEntities.size());
+        assertEquals(1,clusteredVariantEntities.stream().filter(cve->cve.getAccession().equals(3000000003L)).count());
+        assertEquals(3,clusteredVariantEntities.stream().filter(cve->cve.getAccession().equals(3000000001L)).count());
+
+        //assert merge event
+        SubmittedVariantOperationEntity submittedVariantOperationEntityMerge =
+                mongoTemplate.findAll(SubmittedVariantOperationEntity.class).stream()
+                        .filter(svoe->svoe.getEventType().equals(EventType.RS_MERGE_CANDIDATES))
+                        .collect(Collectors.toList()).get(0);
+        assertEquals(2, submittedVariantOperationEntityMerge.getInactiveObjects().size());
+        assertEquals(EventType.RS_MERGE_CANDIDATES, submittedVariantOperationEntityMerge.getEventType());
+        assertEquals(3000000003L,submittedVariantOperationEntityMerge.getAccession().longValue());
+        //assertEquals("RS mismatch between variants", submittedVariantOperationEntityMerge.getReason());
+
+        //assert split event
+        SubmittedVariantOperationEntity submittedVariantOperationEntitySplit =
+                mongoTemplate.findAll(SubmittedVariantOperationEntity.class).stream()
+                        .filter(svoe->svoe.getEventType().equals(EventType.RS_SPLIT_CANDIDATES))
+                        .collect(Collectors.toList()).get(0);
+        assertEquals(3, submittedVariantOperationEntitySplit.getInactiveObjects().size());
+        assertEquals(EventType.RS_SPLIT_CANDIDATES, submittedVariantOperationEntitySplit.getEventType());
+        assertEquals(3000000001L,submittedVariantOperationEntitySplit.getAccession().longValue());
+        assertEquals("Hash mismatch with 3000000001", submittedVariantOperationEntitySplit.getReason());
```

Comment:
Since this test and the previous one share common assertions, I suggest refactoring them to a separate function so that we are explicit about testing the same set of assertions in multi-batch and single-batch scenarios.

Reply:
moved to a method